akka-actor-typed/src/main/scala/akka/actor/typed/internal/InterceptorImpl.scala

```diff
@@ -78,8 +77,31 @@ private[akka] final class InterceptorImpl[O, I](val interceptor: BehaviorInterce
   }
 
   override def receiveSignal(ctx: typed.TypedActorContext[O], signal: Signal): Behavior[O] = {
-    val interceptedResult = interceptor.aroundSignal(ctx, signal, signalTarget)
-    deduplicate(interceptedResult, ctx)
+    if (signal == PostStop) {
+      nestedBehavior match {
+        case s: StoppedBehavior[O] ⇒
+          s.postStop match {
+            case OptionVal.Some(callback) ⇒
+              interceptor.aroundSignal(ctx, PostStop, new SignalTarget[I] {
+                def apply(ctx: TypedActorContext[_], signal: Signal): Behavior[I] = {
+                  callback match {
+                    case x: ExtensibleBehavior[I] ⇒
```

Comment:
always ExtensibleBehavior?

Reply:
Always `ReceiveBehavior` (because the only use case is `Behaviors.receiveSignal`) which extends `ExtensibleBehavior`, this should cover `AbstractBehavior` as well. (I'd want to narrow that down to being an impl detail though, see #26293 )